base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/views/ScriptGeneratorViewModel.java

```diff
@@ -730,11 +747,75 @@ public void reloadActions() {
 	}
 	
 	/**
-	 * Get the user manual URL for the script generator from the model.
+	 * Save Parameter Values to a file. Checks if file already exists and asks if user wants to overwrite it.
+	 */
+	public void saveParameterValues() {		
+		SaveParametersDialog dialog = new SaveParametersDialog(null);
+		if (dialog.open() == Window.OK) {
+			String fileName = dialog.getFileName();
+			try {
+				boolean userWantsToOverwrite = true;
+				List<String> listOfDataFiles = scriptGeneratorModel.getListOfdataFiles();
+				if (listOfDataFiles.stream().anyMatch(fileName::equalsIgnoreCase)) {
+					userWantsToOverwrite = MessageDialog.openConfirm(DISPLAY.getActiveShell(), "Duplicate Name" ,String.format("File name %s already exist\nDo you want to overwrite it?", fileName));
+				}
+				if (userWantsToOverwrite) {
+					scriptGeneratorModel.saveParameters(fileName);
+					MessageDialog.openInformation(DISPLAY.getActiveShell(), "Saved", "File saved!");
+				}
+			} catch (NoScriptDefinitionSelectedException e) {
+				LOG.error(e);
+				MessageDialog.openWarning(DISPLAY.getActiveShell(), "No script definition selection", 
+						"Cannot generate script. No script definition has been selected");
+			} catch (FileNotFoundException e) {
+				MessageDialog.openWarning(DISPLAY.getActiveShell(), "C:/ScriptGeneratorDataFiles not found ","Folder to save data files not found");
+			}
+		}
+		 
+	}
+	
+	/**
+	 * Load parameter values.
+	 */
+	public void loadParameterValues() {
+		DataFileSelectionDialog dialog = new DataFileSelectionDialog(Display.getDefault().getActiveShell(),
+				"list of files", this.getListOfDataFiles());
+		if (dialog.open() == Window.OK) {
+			String selectedFile = dialog.selectedFile();
+			try {
+				scriptGeneratorModel.loadParameterValues(selectedFile);
+			} catch (NoScriptDefinitionSelectedException e) {
+				LOG.error(e);
+				MessageDialog.openWarning(DISPLAY.getActiveShell(), "No script definition selection", 
+						"Cannot generate script. No script definition has been selected");
+			} catch (ScriptDefinitionNotMatched | UnsupportedOperationException e) {
+				LOG.error(e);
+				MessageDialog.openError(DISPLAY.getActiveShell(), "Error", e.getMessage());
+			}
+		}
+		
+	};
+		
+	/**
+	 * Get list of data files.
+	 * @return list of data files
+	 */
+	public List<String> getListOfDataFiles() {
+		try {
+			return scriptGeneratorModel.getListOfdataFiles();
+		} catch (FileNotFoundException e) {
+			MessageDialog.openError(Display.getDefault().getActiveShell(), "Error", "C:/DataFile folder where parameters values will be saved"
```

Comment:
This isn't the correct path, also it shouldn't be hardcoded, you'll need to get it's value from the preference store

Reply:
Removed as it is not relevant anymore.